packages/m2m-gateway/src/services/eserviceService.ts

```diff
@@ -664,15 +662,71 @@ export function eserviceServiceBuilder(
         headers
       );
 
-      const createdRiskAnalysis = eservice.riskAnalysis.find(
-        (r) => r.id === createdRiskAnalysisId
+      const createdRiskAnalysis = retrieveEServiceRiskAnalysisById(
+        { data: eservice, metadata },
+        unsafeBrandId(createdRiskAnalysisId)
       );
 
-      if (!createdRiskAnalysis) {
-        throw eserviceRiskAnalysisNotFound(eserviceId, createdRiskAnalysisId);
-      }
-
       return toM2MGatewayApiEServiceRiskAnalysis(createdRiskAnalysis);
     },
+
+    async getEServiceRiskAnalyses(
+      eserviceId: EServiceId,
+      { limit, offset }: m2mGatewayApi.GetEServiceRiskAnalysesQueryParams,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceRiskAnalyses> {
+      logger.info(`Retrieving Risk Analyses for E-Service ${eserviceId}`);
+
+      const { data: eservice } = await retrieveEServiceById(
+        headers,
+        eserviceId
+      );
+
+      const paginated = eservice.riskAnalysis.slice(offset, offset + limit);
```

Comment:
I did this with `slice` since it's quick and we are close to the release (we did the same for purpose versions and other sub-resources). In the future, we might want to move the listing to the process with pagination leveraged at the DB query level - tracked here: [PIN-7463](https://pagopa.atlassian.net/browse/PIN-7463)